drivers/video/hm0360.c

```diff
@@ -0,0 +1,925 @@
+			      (struct video_ctrl_range){.min = 0, .max = 1, .step = 1, .def = 0});
+	if (ret) {
+		return ret;
+	}
+
+	return 0;
+}
+
+static int hm0360_init(const struct device *dev)
+{
+	const struct hm0360_config *config = dev->config;
+	struct video_frmival frmival;
+
+	int ret;
+
+	struct video_format fmt = {
+		.pixelformat = VIDEO_PIX_FMT_GREY,
+		.width = 320,
+		.height = 240,
+	};
+
+	if (!i2c_is_ready_dt(&config->bus)) {
+		/* I2C device is not ready, return */
+		return -ENODEV;
+	}
+
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(pwdn_gpios)
+	/* Power up camera module */
+	if (config->pwdn.port != NULL) {
+		if (!gpio_is_ready_dt(&config->pwdn)) {
+			return -ENODEV;
+		}
+		ret = gpio_pin_configure_dt(&config->pwdn, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not clear power down pin: %d", ret);
+			return ret;
+		}
+	}
+	gpio_pin_set_dt(&config->pwdn, 1);
+#endif
+#if DT_ANY_INST_HAS_PROP_STATUS_OKAY(reset_gpios)
+	/* Reset camera module */
+	if (config->reset.port != NULL) {
+		if (!gpio_is_ready_dt(&config->reset)) {
+			return -ENODEV;
+		}
+		ret = gpio_pin_configure_dt(&config->reset, GPIO_OUTPUT_ACTIVE);
+		if (ret < 0) {
+			LOG_ERR("Could not set reset pin: %d", ret);
+			return ret;
+		}
+		/* Reset is active high, has 1ms settling time*/
+		k_msleep(1);
+		gpio_pin_set_dt(&config->reset, 0);
+		k_msleep(1);
+	}
+#endif
+
+	/*
+	 * Read product ID from camera. This camera implements the SCCB,
+	 * spec- which *should* be I2C compatible, but in practice does
+	 * not seem to respond when I2C repeated start commands are used.
+	 * To work around this, use a write then a read to interface with
+	 * registers.
+	 */
+	uint32_t Data;
+	uint16_t pid = 0x0000;
+
+	/*  Read Product ID and check if PID OK  */
+	ret = video_read_cci_reg(&config->bus, HM0360_REG_A16D8(HM0360_MODEL_ID_H), &Data);
+	if (ret < 0) {
+		LOG_ERR("Could not read product ID: %d", ret);
+		return ret;
+	}
+	pid = (Data << 8);
+
+	ret = video_read_cci_reg(&config->bus, HM0360_REG_A16D8(HM0360_MODEL_ID_L), &Data);
+	if (ret < 0) {
+		LOG_ERR("Could not read product ID: %d", ret);
+		return ret;
+	}
+	pid |= Data;
+
+	if (pid != hm0360_PROD_ID) {
+		LOG_ERR("Incorrect product ID: 0x%02X", pid);
+		return -ENODEV;
+	}
+	LOG_ERR("Product ID is: %x", pid);
+
+	/* Reset camera registers */
+	ret = video_write_cci_reg(&config->bus, HM0360_REG_A16D8(HM0360_SW_RESET), HM0360_RESET);
+	if (ret) {
+		LOG_ERR("Could not write reset: %d", ret);
+		return ret;
+	}
+	/* Delay for 1ms. */
+	k_msleep(1);
+
+	/* Write Default registers */
+	ret = video_write_cci_multiregs16(&config->bus, hm0360_default_regs,
+					  ARRAY_SIZE(hm0360_default_regs));
+	if (ret < 0) {
+		LOG_ERR("Failed to set default registers");
+		return ret;
+	}
+
+	hm0360_set_fmt(dev, &fmt);
+	if (ret) {
+		return ret;
+	}
+
+	frmival.numerator = 15;
+	frmival.denominator = 1;
+	if (hm0360_set_frmival(dev, &frmival)) {
+		printk("ERROR: Unable to set up video format\n");
+		return false;
+	}
```

Comment:
You might want to split the `if()` away from the function call so that you can return the original error message, `false == 0` meaning "success".

```suggestion
	ret = hm0360_set_frmival(dev, &frmival);
	if (ret < 0) {
		LOG_ERR("ERROR: Unable to set up video format");
		return ret;
	}
```

Reply:
fixed